draft-ietf-quic-transport.md

```diff
@@ -2678,72 +2678,105 @@ transmission efficiency to underfilled packets.
 
 # Packetization and Reliability {#packetization}
 
-A sender bundles one or more frames in a Regular QUIC packet (see {{frames}}).
+A sender bundles one or more frames in a QUIC packet (see {{frames}}).
 
 A sender SHOULD minimize per-packet bandwidth and computational costs by
 bundling as many frames as possible within a QUIC packet.  A sender MAY wait for
 a short period of time to bundle multiple frames before sending a packet that is
 not maximally packed, to avoid sending out large numbers of small packets.  An
-implementation may use heuristics about expected application sending behavior to
-determine whether and for how long to wait.  This waiting period is an
-implementation decision, and an implementation should be careful to delay
+implementation may use knowledge about application sending behavior or
+heuristics to determine whether and for how long to wait.  This waiting period
+is an implementation decision, and an implementation should be careful to delay
 conservatively, since any delay is likely to increase application-visible
 latency.
 
-Regular QUIC packets are "containers" of frames; a packet is never retransmitted
-whole.  How an endpoint handles the loss of the frame depends on the type of the
-frame.  Some frames are simply retransmitted, some have their contents moved to
-new frames, and others are never retransmitted.
 
-When a packet is detected as lost, the sender re-sends any frames as necessary:
+## Packet Processing and Acknowledgment
 
-* All application data sent in STREAM frames MUST be retransmitted, unless the
-  endpoint has sent a RST_STREAM for that stream.  When an endpoint sends a
-  RST_STREAM frame, data outstanding on that stream SHOULD NOT be retransmitted,
-  since subsequent data on this stream is expected to not be delivered by the
-  receiver.
+A packet MUST NOT be acknowledged until packet protection has been successfully
+removed and all frames contained in the packet have been processed.  Any stream
+state transitions triggered by the frame MUST have occurred.  For STREAM frames,
+this means the data has been enqueued in preparation to be received by the
+application protocol, but it does not require that data is delivered and
+consumed.
+
+Once the packet has been fully processed, a receiver acknowledges receipt by
+sending one or more ACK frames containing the packet number of the received
+packet.  To avoid creating an indefinite feedback loop, an endpoint MUST NOT
+send an ACK frame in response to a packet containing only ACK or PADDING frames,
+even if there are packet gaps which precede the received packet.  The endpoint
+MUST acknowledge packets containing only ACK or PADDING frames in the next ACK
+frame that it sends.
+
+Strategies and implications of the frequency of generating acknowledgments are
+discussed in more detail in {{QUIC-RECOVERY}}.
+
+
+## Retransmission of Information
+
+QUIC packets that are determined to be lost are never retransmitted whole.  The
```

Comment:
Technically, there's the one exception around connection close?

Reply:
I don't like explaining exceptions every place, but that would leave "never", which is a bit absolute.